android-components/components/feature/addons/src/test/java/mozilla/components/feature/addons/ui/AddonFilePickerTest.kt

```diff
@@ -104,4 +106,46 @@ class AddonFilePickerTest {
         val mimeTypes = intent.extras!!.getStringArray(Intent.EXTRA_MIME_TYPES)
         assertArrayEquals(mimeTypes, arrayOf("application/x-xpinstall", "application/zip"))
     }
+
+    @Test
+    fun `WHEN an addon is installed successfully THEN remove the temporary file`() {
+        val uri: Uri = mock()
+
+        doReturn("file:///data/data/XPIs/addon.xpi").`when`(filePicker).convertToFileUri(uri)
+
+        val onSuccessCallbackCapture = argumentCaptor<((Addon) -> Unit)>()
+        filePicker.handleUriSelected(uri)
+
+        verify(addonManager).installAddon(
+            url = any<String>(),
+            installationMethod = eq(InstallationMethod.FROM_FILE),
+            onSuccess = onSuccessCallbackCapture.capture(),
+            onError = any(),
+        )
+
+        onSuccessCallbackCapture.value.invoke(mock())
+
+        verify(filePicker).removeTemporaryFile(any())
```

Comment:
Let's verify that we call `removeTemporaryFile()` with the safe URI instead of `any()`.

Reply:
Updated!